### PR TITLE
[apmpackage] Add config options supported in ESS

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
Add config options to `apmpackage` for currently supported config options on ESS. This only adds options that make sense to support when run by Elastic Agent. There is currently no out-of-the-box way to group the options, although it would be nice to introduce e.g. a `rum` and `security` section. 

@jalvz and @axw I'd appreciate feedback regarding the wording and the config options themselves. 

Currently supported in ESS, that are left out:
not supported by Agent:
* `apm-server.register.ingest.pipeline.*`
* `apm-server.rum.source_mapping.*`
* `apm-server.ilm`
* `setup.template` 
* `logging.*` (only `logging.level` is supported - per agent, not per input though)

not deemed to be necessary:
* `apm-server.agent.config.cache.expiration`
* `max_procs`

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes
* Check out the APM-Integration-Testing and run `./scripts/compose.py start master --with-elastic-agent --no-apm-server --with-package-registry --package-registry-apm-path=<your-path>/apm-server/apmpackage/apm` 
* Try to add the integration to the default policy

## Related issues
part of #4528

## Screenshot
<details>
<summary>Config options - Basic</summary>
Screenshot 2021-02-15 at 15 38 40

</details>

<details>
<summary>Config options - Advanced </summary>
Screenshot 2021-02-15 at 15 38 49

</details>

### CODE REVIEW

File: apmpackage/apm/0.1.0/manifest.yml [hunk 1]
```diff
@@ -35,12 +35,94 @@ policy_templates:
...
+        required: false
+        show_user: false
+      - name: rum_exclude_from_grouping
+        type: string
+        title: RUM - pattern excluding frames from grouping
+        required: false
+        show_user: false
+      - name: rum_response_headers
+        type: yaml
+        title: RUM - Custom Response Headers
```
Comment (axw) [inline apmpackage/apm/0.1.0/manifest.yml:87]:
```suggestion
        title: RUM - Custom response headers
```

File: apmpackage/apm/0.1.0/manifest.yml [hunk 2]
```diff
@@ -35,12 +35,94 @@ policy_templates:
...
+      - name: rum_library_pattern
+        type: string
+        title: RUM - pattern identifying library frames
+        required: false
+        show_user: false
+      - name: rum_exclude_from_grouping
+        type: string
+        title: RUM - pattern excluding frames from grouping
+        required: false
+        show_user: false
```
Comment (axw) [inline apmpackage/apm/0.1.0/manifest.yml:84]:
```suggestion
```

I know we discussed adding these, but I'm having second thoughts. Let's add them in once we've added support for sourcemaps?

File: apmpackage/apm/0.1.0/manifest.yml [hunk 3]
```diff
@@ -35,12 +35,94 @@ policy_templates:
...
         default: false
+      - name: api_key_limit
+        type: int
+        title: Maximum number of allowed API Keys
+        required: false
+        show_user: false
+        default: 100
+      - name: capture_personal_data
+        type: bool
+        title: Capture Personal Data (e.g. IP, User Agent)
```
Comment (axw) [inline apmpackage/apm/0.1.0/manifest.yml:58]:
```suggestion
        title: Capture personal data (e.g. IP, User Agent)
```

**Review by axw — COMMENTED** [review]
Looks good overall, mostly just nit-picking.

As we only support one kind of buffering I'd prefer to say "Event buffer" rather than "In-Memory buffer".

It would be nice to add descriptions to all of these, particularly the likes of "RUM - Allow Origins". Perhaps in a followup?

**Comment by axw** [comment]
We should get a designer involved at some point, but the Endpoint Security policy editor is nice:

image

For RUM, ideally we would have something like this. A radio button for enabling/disabling RUM, and then RUM-related config inside a box.

File: apmpackage/apm/0.1.0/manifest.yml (hunk 2 above)
Comment (simitt) [reply apmpackage/apm/0.1.0/manifest.yml:84]: SGTM

File: apmpackage/apm/0.1.0/manifest.yml [hunk 4]
```diff
@@ -35,12 +35,84 @@ policy_templates:
         title: Secret token
         required: false
         show_user: true
+      - name: api_key_enabled
+        type: bool
+        title: Enable API Key Auth from APM Agents
```
Comment (jalvz) [inline apmpackage/apm/0.1.0/manifest.yml:40]:
```suggestion
        title: Enable API Key Auth between APM Server and Agents
```

File: apmpackage/apm/0.1.0/manifest.yml [hunk 5]
```diff
@@ -35,12 +35,84 @@ policy_templates:
...
+        default: false
       - name: enable_rum
         type: bool
         title: Enable RUM
         required: true
         show_user: true
         default: false
+      - name: api_key_limit
+        type: int
+        title: Maximum number of allowed API Keys
```
Comment (jalvz) [inline apmpackage/apm/0.1.0/manifest.yml:52]:
```suggestion
        title: Maximum number of allowed API Keys between APM Server and Agents
```

We'll possibly have more api keys in the config, better to be explicit

File: apmpackage/apm/0.1.0/agent/input/template.yml.hbs [hunk 6]
```diff
@@ -2,3 +2,17 @@ apm-server:
...
+    api_key.enabled: {{api_key_enabled}}
+    api_key.limit: {{api_key_limit}}
+    rum.allow_origins: {{rum_allow_origins}}
+    rum.allow_headers: {{rum_allow_headers}}
+    rum.library_pattern: {{rum_library_pattern}}
+    rum.exclude_from_grouping: {{rum_exclude_from_grouping}}
+    rum.response_headers: {{rum_response_headers}}
+    rum.event_rate.limit: {{rum_event_rate_limit}}
+    rum.event_rate.lru_size: {{rum_event_rate_lru_size}}
+queue.mem.events: {{queue_mem_events}}
```
Comment (jalvz) [inline apmpackage/apm/0.1.0/agent/input/template.yml.hbs:16]: Did you test that `queue.*` actually get picked up? I would not expect this to work, but I'm glad if it does.

**Review by jalvz — COMMENTED** [review]
FYI there is mention of settings in https://github.com/elastic/apm-server/blob/master/apmpackage/apm/0.1.0/_dev/docs/README.template.md#configuration-parameters, either add all of these or remove the section altogether.

File: apmpackage/apm/0.1.0/agent/input/template.yml.hbs (hunk 6 above)
Comment (simitt) [reply apmpackage/apm/0.1.0/agent/input/template.yml.hbs:16]: I'll remove the `queue` related pieces; created an issue for reloading the queue when configured for agent https://github.com/elastic/beats/issues/23993.

File: apmpackage/apm/0.1.0/manifest.yml [hunk 7]
```diff
@@ -35,12 +35,84 @@ policy_templates:
...
+        required: false
+        show_user: false
+        default: true
+      - name: rum_allow_origins
+        type: string
+        title: RUM - Allow Origins
+        multi: true
+        required: false
+        show_user: false
+        default: ['*']
```
Comment (simitt) [inline apmpackage/apm/0.1.0/manifest.yml:68]: filed a bug report https://github.com/elastic/kibana/issues/91401, as this value is currently not allowed.

**Comment by simitt** [comment]
@axw fyi - created https://github.com/elastic/kibana/issues/91412 for the APM Server configuration editor

**Review by jalvz — APPROVED** [review]
LGTM, but do you mind removing https://github.com/elastic/apm-server/blob/master/apmpackage/apm/0.1.0/_dev/docs/README.template.md#configuration-parameters if you are not updating it? We can leave that for @bmorelli25,  he'll figure out how best to document that :) 

Perhaps, also add that the queue settings are not supported in this version in the "Limitations" section.

**Comment by simitt** [comment]
@jalvz sorry I forgot about that, pushed it now [381a80c](https://github.com/elastic/apm-server/pull/4690/commits/381a80c9b3036011ba6c95a81d57215c7483ef7a). I also removed the line about reloading, since that should be supported already.

**Comment by jalvz** [comment]
Tested trough the UI, works as expected

**Pull request merged**